apps/rush-lib/src/cli/actions/GenerateAction.ts

```diff
@@ -110,21 +119,24 @@ export default class GenerateAction extends BaseRushAction {
       }
     } catch (ex) {
       console.log();
-      console.log('There was a problem reading the shrinkwrap file. Proceeeding with "rush generate".');
+      console.log('There was a problem reading the shrinkwrap file. Proceeding with "rush generate".');
     }
 
     return installManager.ensureLocalPackageManager(false).then(() => {
       installManager.createTempModules(true);
 
-      // Delete both copies of the shrinkwrap file
-      if (fsx.existsSync(committedShrinkwrapFilename)) {
-        console.log(os.EOL + 'Deleting ' + committedShrinkwrapFilename);
-        fsx.unlinkSync(committedShrinkwrapFilename);
-      }
-      if (fsx.existsSync(tempShrinkwrapFilename)) {
-        fsx.unlinkSync(tempShrinkwrapFilename);
+      if (this._conservativeParameter.value) {
+        if (fsx.existsSync(committedShrinkwrapFilename)) {
+          console.log(os.EOL + 'Preserving ' + committedShrinkwrapFilename);
+        } else {
+          console.log(os.EOL + 'Missing ' + committedShrinkwrapFilename);
```

Comment:
>nsole.log(os.EOL + 'Missing ' + committedShrinkwrapFilename [](start = 12, length = 59)

If you are missing the committedShrinkwrapFile, does doing a `--conservative` generate even make sense?

Reply:
No, fixed

---
In reply to: [171737620](https://github.com/Microsoft/web-build-tools/pull/549#discussion_r171737620) [](ancestors = 171737620)